aws/fsx.go

```diff
@@ -89,3 +109,20 @@ func waitForFsxFileSystemUpdate(conn *fsx.FSx, id string, timeout time.Duration)
 
 	return err
 }
+
+func waitForFsxFileSystemUpdateOptimizing(conn *fsx.FSx, id string, timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{fsx.StatusInProgress},
```

Comment:
Seems like FSx also uses a `PENDING` status before it goes in progress. Created followup issue: https://github.com/hashicorp/terraform-provider-aws/issues/16440